### ISSUE TITLE
Popover complex content not read by Voice Over

### ISSUE THREAD

**Issue by furstenberg** [sl-design-system/components#972]
**Describe the bug**

I navigated to the Popover component in [Storybook](https://kind-plant-088739303.1.azurestaticapps.net/?path=/story/popover--more-complex-content&globals=theme:itslearning) using keyboard and Voice Over. I toggled the button, made aware by Voice Over that the button was in a collapsed state. The focus then shifted to a Cancel Button, with no information to me what I should Cancel.

By continuing to press TAB I landed on a Confirm button, which indicates that some sort of dialog has been presented to me. Contents of this dialog was not read by Voice Over.  By pressing TAB once more I landed away from the dialog, which was confusing.

**To Reproduce**

1. Launch Voice Over
2. Navigate to Button on the Popover - More Complex Content page
3. Press Enter when you've landed on the Button
4. Have Voice Over read button for you
5. Press TAB
6. Have Voice Over read button for you
7. Wonder what I want to cancel or confirm

**Expected behavior**

- I expect the contents of the dialog to be read when I toggle the button. So I can understand what they want me to answer Cancel or Confirm to.
- I expect a "tab trap" to be implemented so I cannot navigate outside the dialog.

**Screenshots**

https://github.com/sl-design-system/components/assets/18802/bd7b6b3c-9342-4c8b-bbbb-aa840b1da744

In the Voice Over output window you can see exactly what I heard.

**Desktop (please complete the following information):**
 - OS: [e.g. macOS 14.3.1]
 - Browser [Latest stable Brave, Safari and Firefox]

**Additional context**

I understand the component is not done yet. So if this feedback is "too early", please just close this. :-)

**Comment by jpzwarte** [comment]
@furstenberg I'm looking at some online stuff about this and found this: https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/examples/disclosure-navigation/

From what i see, it looks like common practice it to assign `role="region"` to `<sl-popover>` and perhaps also `aria-labelledby" and have that point to the title of the popover. As far as i can tell, having only `aria-expanded` on the button seems enough. What do you think?

**Comment by furstenberg** [comment]
The disclosure pattern is typically used more for an "accordion" style of UI. And I don't see how that would change anything here.

To me there are two different types of issues here. 

One is that the "Basic" Popup doesn't read the contents when using a screen reader. That can perhaps be solved by using `aria-describedby`.  But not all screen reader likes when attributes are added after the initial rendering.

The other issue, is that I don't see any difference between this dialog and a regular dialog launched from a button. So why not use a regular button to launch a dialog?

**Comment by jpzwarte** [comment]
Feedback from hidde: `aria-details` is on the wrong element.
